hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperationProviderDelegator.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation;
+
+import com.hazelcast.core.EntryView;
+import com.hazelcast.map.EntryProcessor;
+import com.hazelcast.map.impl.MapEntries;
+import com.hazelcast.map.merge.MapMergePolicy;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.spi.OperationFactory;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Base class which basically delegates all method calls to underlying {@link MapOperationProvider}
+ * to prevent extending classes to override all methods.
+ *
+ * See {@link WANAwareOperationProvider} for an example.
+ */
+public abstract class MapOperationProviderDelegator implements MapOperationProvider {
+
+    abstract MapOperationProvider getOperationProviderDelegate();
```

Comment:
Maybe renamed as `getDelegate()`, we know this is map-operation-provider from context.